source/clustering.md

```diff
@@ -374,37 +444,117 @@ we sum them together to get the *total WSSD*.
 For our example,
 this means adding up all the squared distances for the 18 observations.
 These distances are denoted by black lines in
-{numref}`toy-example-all-clus-dists-1`.
+{numref}`toy-example-all-clus-dists`
 
-```{figure} img/clustering/toy-example-all-clus-dists-1.png
----
-height: 400px
-name: toy-example-all-clus-dists-1
----
-All clusters from the penguin_data data set example. Observations are in orange, blue, and yellow with the cluster center highlighted in red. The distances from the observations to each of the respective cluster centers are represented as black lines.
+:::{glue:figure} toy-example-all-clus-dists
+:figwidth: 700px
+:name: toy-example-all-clus-dists
 
+All clusters from the `penguin_data` data set example. Observations are in blue, orange, and red with the cluster center highlighted in orange. The distances from the observations to each of the respective cluster centers are represented as black lines.
+:::
 
-```
++++
 
 ### The clustering algorithm
 
 ```{index} K-means; algorithm
 ```
 
+```{code-cell} ipython3
+:tags: [remove-cell]
+
+penguin_data = pd.read_csv("data/penguins_standardized.csv")
+# Set up the initial "random" label assignment the same as in the R book
+penguin_data['label'] = [
+    2, 2, 1, 1, 0, 0, 0, 1,
+    2, 2, 1, 2, 1, 2,
+    0, 1, 2, 2
+]
+points_kmeans_init = alt.Chart(penguin_data).mark_point(size=75, filled=True, opacity=1).encode(
+    alt.X("flipper_length_standardized").title("Flipper Length (standardized)"),
+    alt.Y("bill_length_standardized").title("Bill Length (standardized)"),
+    alt.Color('label:N').legend(None),
+    alt.Shape('label:N').legend(None).scale(range=['square', 'circle', 'triangle']),
+    alt.Size('label:O').legend(None).scale(type='ordinal', range=[50, 50, 100]),
```

Comment:
The reason for the different sizes is that triangle marks seem to be smaller by default and hard to see